src/Analyser/NodeScopeResolver.php

```diff
@@ -1465,6 +1465,24 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			) {
 				$scope = $scope->assignVariable('http_response_header', new ArrayType(new IntegerType(), new StringType()));
 			}
+
+			if (
+				isset($functionReflection)
+				&& $functionReflection->getName() === 'extract'
+				&& $scope->getType($expr->args[0]->value) instanceof ConstantArrayType
+				&& in_array(count($expr->args), [1, 2], true) // currently only understands simple extract usage
+			) {
+				$flags = $scope->getType($expr->args[1]->value);
+
+				if ($flags instanceof ConstantIntegerType && $flags->getValue() === 0) {
+					$arrayArg = $scope->getType($expr->args[0]->value);
+
+					array_map(static function (ConstantStringType $k, $v) use (&$scope): void {
```

Comment:
I don't really like this usage of array_map (with `void`) - array map is supposed to transform the array. I'd much rather like a `foreach` here.

Reply:
You should at least differentiate between: missing second arg (same as OVERWRITE), having `OVERWRITE` there, having `SKIP` there.

Currently, you only support `OVERWRITE`. Supporting `SKIP` and the PREFIX variants should be there too. If you ask with bitwise mechanics, it's really easy. (`$flags->getValue() & EXTR_SKIP === EXTR_SKIP`). It's just a question of a few if branches.

I'm pretty sure your current code will crash when the function is called without second argument (`extract($vars)`) - this combination isn't tested.

Reply:
Thanks for your feedback and your patience :-)
I'll need some time to get this right, but I'm on it.

(I don't want to waste your time, and I'll write many more tests to do so.)